src/services/FileService.ts

```diff
@@ -0,0 +1,271 @@
+    return await this.getMinioService().getDownloadPresignedUrl(this.bucketName, objectKey);
+  }
+
+  /**
+   * Create a new file record
+   */
+  async createFile(
+    name: string,
+    objectKey: string,
+    path: string,
+    size: number,
+    userId: number,
+    mimeType?: string
+  ): Promise<File> {
+    const { services } = await this.getServices();
+    
+    // Validate path format
+    this.validatePath(path);
+    
+    const file = new File();
+    file.name = name;
+    file.objectKey = objectKey;
+    file.path = path;
+    file.size = size;
+    file.userId = userId;
+    if (mimeType) {
+      file.mimeType = mimeType;
+    }
+    
+    await services.em.persistAndFlush(file);
+    return file;
+  }
+
+  /**
+   * Get file by ID and user ID
+   */
+  async getFileById(fileId: number, userId: number): Promise<File | null> {
+    const { services } = await this.getServices();
+    
+    return await services.file.findOne({ id: fileId, userId });
+  }
+
+  /**
+   * Update file (name and path only - virtual changes)
+   */
+  async updateFile(
+    fileId: number, 
+    userId: number, 
+    name?: string, 
+    path?: string
+  ): Promise<File | null> {
+    const { services } = await this.getServices();
+    
+    const file = await this.getFileById(fileId, userId);
+    if (!file) {
+      return null;
+    }
+
+    if (name !== undefined) {
+      file.name = name;
+    }
+
+    if (path !== undefined) {
+      this.validatePath(path);
+      file.path = path;
+    }
+
+    await services.em.flush();
+    return file;
+  }
+
+  /**
+   * Delete file (removes from S3 and database)
+   */
+  async deleteFile(fileId: number, userId: number): Promise<boolean> {
+    const { services } = await this.getServices();
+    
+    const file = await this.getFileById(fileId, userId);
+    if (!file) {
+      return false;
+    }
+
+    try {
+      // Delete from S3
+      await this.getMinioService().deleteFile(this.bucketName, file.objectKey);
+      
+      // Delete from database
+      await services.em.removeAndFlush(file);
+      return true;
+    } catch (error) {
+      console.error('Error deleting file:', error);
+      return false;
+    }
+  }
+
+  /**
+   * List files by directory with pagination
+   */
+  async listFiles(
+    userId: number,
+    path?: string,
+    page: number = 1,
+    limit: number = 20
+  ): Promise<Page<File>> {
+    const { services } = await this.getServices();
+    
+    const offset = (page - 1) * limit;
+
+    // Build where clause
+    const whereClause: any = { userId };
+    if (path) {
+      whereClause.path = path;
+    }
+
+    const findAndCount = await services.file.findAndCount(whereClause, {
+      orderBy: { createdAt: 'DESC' },
+      limit,
+      offset,
+    });
+
+    return toPageDTO(findAndCount, page, limit);
+  }
+
+  /**
+   * Get all files in a directory tree (recursive)
+   */
+  async getDirectoryTree(userId: number, path: string = '/'): Promise<File[]> {
```

Comment:
This will return files in a nested folders too. Eg path /foo have bar.txt and food-bar/ folder this api will return all files in foo-bar/ folder in flat list try to normilize data so frontend now that /foo-bar/ is a folder to fetch files and folder in foo-bar/ folder must pass in the path params is /foo/foo-bar/